src/main/java/hudson/plugins/sidebar_link/ProjectLinks.java

```diff
@@ -40,7 +40,7 @@
  * Add links in a job page sidepanel.
  * @author Alan Harder
  */
-public class ProjectLinks extends JobProperty<AbstractProject<?,?>> {
+public class ProjectLinks extends JobProperty<Job<?,?>> {
```

Comment:
What exactly is your concern here? There are no OSS plugins dependent on sidebar-link

Sam mentioned the fix would just be keeping the old methods that use the type parameter  (cc @svanoort, correct me if I'm wrong)